service/windows/src/node.rs

```diff
@@ -0,0 +1,478 @@
+use crate::subprocess::send_child_ctrl_break;
+use log::*;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{Read, Seek, SeekFrom, Write};
+use std::iter::FromIterator;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use std::os::windows::process::CommandExt;
+use std::path::PathBuf;
+use std::process::{Child, Command, ExitStatus, Stdio};
+use tempfile::{NamedTempFile, TempPath};
+use toml::Value;
+use winapi::um::winbase::CREATE_NEW_PROCESS_GROUP;
+
+/// Default RPC port
+const DEFAULT_RPC_PORT: u16 = 10000;
+
+pub struct LogFileConfig {
+    pub path: PathBuf,
+    pub roll_size: Option<String>,
+    pub roll_count: Option<u16>,
+    pub roll_pattern: Option<String>,
+}
+
+pub enum LoggerConfig {
+    NoLog,
+    ConfigLog(PathBuf),
+    FileLog(LogFileConfig),
+}
+
+/// Runner for a node
+pub struct NodeConfig {
```

Comment:
I guess it could've been preferred if we could have been using the already existing configuration type(s) as they seem quite identical? Maybe the `NodeConfig` could just wrap the needed cfg types from `configuration.rs`?

Reply:
Perhaps, but I think there are enough meaningful differences that having a new configuration type makes sense. I think it would be a bit of an awkward refactor to completely share the configuration `struct`s.